src/rosdep2/main.py

```diff
@@ -918,6 +930,87 @@ def command_resolve(args, options, lookup=None):
         return 1  # error exit code
 
 
+def command_search(args, options):
+    os_override = convert_os_override_option(options.os_override)
+    sources_loader = SourcesListLoader.create_default(sources_cache_dir=options.sources_cache_dir,
+                                                      os_override=os_override,
+                                                      verbose=options.verbose)
+    if FALL_BACK_TO_RE:
+        print("python3-regex module not available, falling back to re module without fuzzy search.")
+        regexes = args
+        regex_flags = re.IGNORECASE
+    else:
+        # Turn search args into regexes to allow for fuzzy search with 2 mistake
+        regexes = ["(?:%s){e<=%s}" % (re.escape(arg), 2 if len(arg) >= 7 else 1) for arg in args]
+        regex_flags = re.BESTMATCH | re.IGNORECASE
+    if options.verbose:
+        print("Searching using the following regexes: %s" %regexes)
+    regexes = [re.compile(r, regex_flags) for r in regexes]
+    close_keys = []
+    close_pkgs = []
+    for view_name in sources_loader.get_loadable_views():
+        view = sources_loader.get_source(view_name=view_name)
+        if not isinstance(view, CachedDataSource):
+            if options.verbose:
+                print('Skipping non-cached source %s' % view_name)
```

Comment:
Can we tell users what to do in this case?

Reply:
Honestly, I don't know what to do there.
I'm skipping them because I don't know the structure if they're not this type.
This print is purely for debugging purposes.
I think a typical installation wouldn't encounter it, at least I never did.